src/interface/generator.ts

```diff
@@ -14,16 +14,35 @@ export interface LabelValueType {
   value?: RawValueType;
   label?: React.ReactNode;
 }
-export type DefaultValueType = RawValueType | RawValueType[] | LabelValueType | LabelValueType[];
+export type DefaultValueType =
+  | RawValueType
+  | RawValueType[]
+  | LabelValueType
+  | LabelValueType[];
 
 export interface DisplayLabelValueType extends LabelValueType {
   disabled?: boolean;
 }
 
-export type SingleType<MixType> = MixType extends (infer Single)[] ? Single : MixType;
+export type SingleType<MixType> = MixType extends (infer Single)[]
+  ? Single
+  : MixType;
 
 export type OnClear = () => void;
 
+export interface GetTagCloseProps {
+  (): {
+    onClick?: (event: React.MouseEvent<HTMLSpanElement, MouseEvent>) => void;
+    closable: boolean;
+  };
+}
+
+export type CustomTagProps = {
+  label: DefaultValueType;
+  disabled: boolean;
+  getTagCloseProps: GetTagCloseProps;
```

Comment:
Should flatten as other props instead of a function.

Reply:
Commit a11ed8829875b6e844f93e2382176c59b592f66f bring some changes to that. Would the fit better the overall code ?